### PR TITLE
feat(oas): support webhooks from oas3.1

### PR DESCRIPTION
**Pull request by daniel-white**
BREAKING CHANGE

adds support for oas3.1 webhooks

addresses #2492

### CODE REVIEW

File: packages/elements-core/src/components/TableOfContents/TableOfContents.tsx [hunk 1]
```diff
@@ -34,6 +34,7 @@ import {
 
 const ActiveIdContext = React.createContext<string | undefined>(undefined);
 const LinkContext = React.createContext<CustomLinkComponent | undefined>(undefined);
+LinkContext.displayName = 'LinkContext';
```
Comment (daniel-white) [inline packages/elements-core/src/components/TableOfContents/TableOfContents.tsx:37]: trying to make this work easier with the dev tools

File: packages/elements-core/src/components/TableOfContents/TableOfContents.tsx [hunk 2]
```diff
@@ -261,8 +265,9 @@ const Group = React.memo<{
         depth={depth}
         isActive={showAsActive}
         icon={
-          NODE_TITLE_ICON[item.title] && (
```
Comment (daniel-white) [inline packages/elements-core/src/components/TableOfContents/TableOfContents.tsx:264]: when displaying the zoom example spec, this was causing an endpoint group to have this icon - so added a type prop (not sure if dev portal's toc needs to include this or not)

File: packages/elements-core/src/components/TableOfContents/types.ts [hunk 3]
```diff
@@ -29,14 +29,17 @@ export type TableOfContentsGroupItem =
 export type TableOfContentsGroup = {
   title: string;
   items: TableOfContentsGroupItem[];
+  itemsType?: 'article' | 'http_operation' | 'http_webhook' | 'model';
```
Comment (daniel-white) [inline packages/elements-core/src/components/TableOfContents/types.ts:32]: 🤔 does the ToC api need to set this as well?

Comment (mallachari) [reply packages/elements-core/src/components/TableOfContents/types.ts:32]: In a way yes. There a `type` property in `TableOfContentsNode` that we'll need to extend with `http_webhook` type

**Review by mallachari — APPROVED** [review]
Code and UI looks great!
Screenshot 2024-01-10 at 01 37 10

Seems like the breaking change applies only to elements-core package.
It might be also worth to add webhook examples to elements-core storybook HttpOperation and TableOfContents stories. And/or adding new story to elements package storybook with simple document with webhooks.

**Review by brendarearden — APPROVED** [review]
LGTM!

**Review by EdVinyard — APPROVED** [review]

**Pull request merged**